test/check-machines-nics

```diff
@@ -242,16 +248,20 @@ class TestMachinesNICs(VirtualMachinesCase):
         b.click("#vm-subVmTest1-run")
         b.wait_in_text("#vm-subVmTest1-state", "Running")
 
-        b.click("#vm-subVmTest1-network-2-edit-dialog")
+        open(2)
         b.select_from_dropdown("#vm-subVmTest1-network-2-select-source", "eth42")
         b.wait_attr("#vm-subVmTest1-network-2-edit-dialog-mac", "readonly", "")
         b.wait_visible("#vm-subVmTest1-network-2-edit-dialog-idle-message")
-        b.click("#vm-subVmTest1-network-2-edit-dialog-save")
-        b.wait_not_present("#vm-subVmTest1-network-2-edit-dialog-modal-window")
+        save(2)
 
         # The NICs table should display the permanent configuration which is different than the live which we just set
         b.wait_in_text("#vm-subVmTest1-network-2-source", "eth43")
 
+        # Change source type from direct to virtual network - https://bugzilla.redhat.com/show_bug.cgi?id=1977669
+        open(2)
+        b.select_from_dropdown("#vm-subVmTest1-network-2-select-type", "network")
+        save(2)
```

Comment:
This is a pretty shallow test, as it only checks that the UI does not crash. Ideally it should test the actual effect in the virt XML, or in what the "attached networks" UI shows after that.

Reply:
Previously it generated an error in the dialog making the dialog to actually not close, now the dialog closes. I can indeed check that the change took effect.